cell_classification/augmentation_pipeline.py

```diff
@@ -0,0 +1,80 @@
+import imgaug as ia
+import imgaug.augmenters as iaa
+from imgaug.augmentables.segmaps import SegmentationMapsOnImage
+from imgaug.augmentables.batches import Batch
+import numpy as np
+
+
+def augment_images(images, masks, augmentation_pipeline):
+    """
+    Augment images and masks.
+    Args:
+        images (np.array):
+            The images (b,h,w,c) to augment
+        masks (np.array):
+            The masks (b,h,w,1) to augment
+        augmentation_pipeline (imgaug.augmenters.meta.Augmenter):
+            The augmentation pipeline
+    Returns:
+        np.array:
+            The augmented images
+        np.array:
+            The augmented masks
+    """
+    masks = [SegmentationMapsOnImage(mask, shape=images.shape[1:-1]) for mask in masks]
+    batch = Batch(images=images, segmentation_maps=masks)
+    batch = augmentation_pipeline.augment_batch_(batch)
+    augmented_masks = [np.squeeze(mask.arr) for mask in batch.segmentation_maps_aug]
+    return np.stack(batch.images_aug, 0), np.squeeze(np.stack(augmented_masks, 0))
```

Comment:
How come we squeeze the `augmented_masks` twice?

Reply:
This is an error. Should only be squeezed once. The problem is the following: `augmentation_pipeline.augment_batch_(batch)` adds a channel dimension to the mask if there's none, to ensure that input and output dim are the same, I think I should check if a channel dim got added, and if this happened it should be squeezed. Because as it is now, it would also squeeze the batch dim for batch size 1 which is unwanted.

Reply:
Won't the masks always have a channels dim?

We should loop over a couple different shapes for the testing data. Right now it's `[3, 100, 100, 2]`. We should probably check the case where both the first and/or last index is 1.

Reply:
Changed it so that it works for masks with or without channel dim and with different number of channels.